src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -651,6 +661,29 @@ public static bool IsSorted(int[] values)
             return true;
         }
 
+        /// <summary>
+        /// Checks that an input array is monotonically increasing
+        /// </summary>
+        /// <param name="values">An array of values</param>
+        /// <returns>True if the array is monotonically increasing; false otherwise.</returns>
+        public static bool IsSorted(double[] values)
+        {
+            if (Utils.Size(values) <= 1)
+                return true;
+
+            var prev = values[0];
+
+            for (int i = 1; i < values.Length; i++)
+            {
+                if (values[i] < prev)
```

Comment:
what happens in cases of .NAN,  .INF and other special values?  
I believe all comparisons of NAN return false.

Reply:
Special cases include: `Epsilon`, `MaxValue`, `MinValue`, `NaN`, `NegativeInfinity`, `PositiveInfinity`. Only `NaN` has ambiguous behavior. I've noted the expected behavior for this function in the docstring: Anything with a NaN is considered to not be monotonically increasing.